FightsApi/Controllers/VoteController.cs

```diff
@@ -5,25 +5,41 @@
 using System.Threading.Tasks;
 using FightsApi_Models.ViewModels;
 using FightsApi_Buisiness.Interfaces;
+using FightsApi_Models.Exceptions;
 
 namespace FightsApi.Controllers
 {
   [Route("api")]
   [ApiController]
   public class VoteController : Controller
   {
-    private readonly IRepository<ViewVote, int> _vr;
+    private readonly IVoteRepository _vr;
 
-    public VoteController(IRepository<ViewVote, int> vr)
+    public VoteController(IVoteRepository vr)
     {
       _vr = vr;
     }
 
     [HttpPost("/vote")]
     public async Task<ActionResult<ViewVote>> Vote(ViewVote vote)
     {
-      var newVote = await _vr.Add(vote);
-      return Ok(newVote);
+      try
+      {
+        var newVote = await _vr.Add(vote);
+        return Ok(newVote);
+      }
```

Comment:
There should be a catch clause here for all exceptions that we want to return a BadRequest() for.